File: examples/jsimport/examplejs.py
# WARNING: the use of javascript pretty much trashes all chance of using
# pyjamas-desktop.  give serious consideration to doing something OTHER
# than including random bits of javascript off the internet in a pyjamas
# application.  the larger the random bit of javascript, the more chance
# there is that it will interact in some horrendous way with the pyjamas
# infrastructure.
#
# if you ABSOLUTELY MUST use javascript, here's how to do it.
#

from pyjamas import logging
log = logging.getAppendLogger(__name__, logging.DEBUG, logging.PLAIN_FORMAT)

# this simply tells the compiler that the two names are to be dropped
# into the javascript global namespace
from __javascript__ import examplevar, get_examplevar

# the default behaviour of jsimport is to include the javascript file
# "inline" - unmodified - direct into the compiler output
from __pyjamas__ import jsimport

jsimport("example.js")

def main():
    global examplevar 
    #examplevar is actually "out of modules", in a super scope

    log.debug(examplevar)
    examplevar = 'Altered'
    log.debug(get_examplevar())

main()

examplevar = "Realtered"
log.debug(get_examplevar())

File: examples/jsimport/examplejs.js
/* start module: examplejs */
$pyjs.loaded_modules['examplejs'] = function (__mod_name__) {
	if($pyjs.loaded_modules['examplejs'].__was_initialized__) return $pyjs.loaded_modules['examplejs'];
	var $m = $pyjs.loaded_modules["examplejs"];
	$m.__repr__ = function() { return "<module: examplejs>"; };
	$m.__was_initialized__ = true;
	if ((__mod_name__ === null) || (typeof __mod_name__ == 'undefined')) __mod_name__ = 'examplejs';
	$m.__name__ = __mod_name__;


	$m['logging'] = $p['___import___']('pyjamas.logging', null, null, false);
	$m['log'] = $m['logging']['getAppendLogger']((typeof __name__ == "undefined"?$m.__name__:__name__), $p['getattr']($m['logging'], 'DEBUG'), $p['getattr']($m['logging'], 'PLAIN_FORMAT'));
	;
	$m['main'] = function() {

		$m['log']['debug'](examplevar);
		examplevar = 'Altered';
		$m['log']['debug'](get_examplevar());
		return null;
	};
	$m['main'].__name__ = 'main';

	$m['main'].__bind_type__ = 0;
	$m['main'].__args__ = [null,null];
	$m['main']();
	examplevar = 'Realtered';
	$m['log']['debug'](get_examplevar());
	return this;
}; /* end examplejs */


/* end module: examplejs */


/*
PYJS_DEPS: ['pyjamas.logging', 'pyjamas']
*/
/*
PYJS_JS: [('example.js', 'default', 'middle')]
*/
